ert_shared/ensemble_evaluator/prefect_ensemble/prefect_ensemble.py

```diff
@@ -0,0 +1,317 @@
+                        "executable": job.get("executable"),
+                        "args": job.get("args", []),
+                    }
+                    for idx, job in enumerate(step.get("jobs", []))
+                ]
+                table_of_elements.append(
+                    {
+                        "iens": iens,
+                        "stage_name": stage["name"],
+                        "stage_id": self.get_id(iens, stage_name=stage["name"]),
+                        "step_id": self.get_id(
+                            iens, stage_name=stage["name"], step_name=step["name"]
+                        ),
+                        **step,
+                        "jobs": jobs,
+                    }
+                )
+
+        produced = set()
+        ordering = []
+        while table_of_elements:
+            temp_list = produced.copy()
+            for element in table_of_elements:
+                if set(element.get("inputs", [])).issubset(temp_list):
+                    ordering.append(element)
+                    produced = produced.union(set(element["outputs"]))
+                    table_of_elements.remove(element)
+        return ordering
+
+    def get_flow(self, ee_id, dispatch_url, coef_input_files, real_range):
+        with Flow(f"Realization range {real_range}") as flow:
+            for iens in real_range:
+                o_t_res = {}
+                for step in self.get_ordering(iens=iens):
+                    inputs = [o_t_res.get(i, []) for i in step.get("inputs", [])]
+                    stage_task = UnixStep(
+                        resources=[coef_input_files[iens]] + step["resources"],
+                        outputs=step.get("outputs", []),
+                        job_list=step.get("jobs", []),
+                        iens=iens,
+                        cmd="python3",
+                        url=dispatch_url,
+                        step_id=step["step_id"],
+                        stage_id=step["stage_id"],
+                        ee_id=ee_id,
+                        on_failure=partial(self._on_task_failure, url=dispatch_url),
+                        run_path=self.config.get("run_path"),
+                        storage_config=self.config.get("storage"),
+                        max_retries=self.config.get("max_retries", 2),
+                        retry_delay=timedelta(seconds=2),
+                    )
+                    result = stage_task(expected_res=inputs)
+
+                    for o in step.get("outputs", []):
+                        o_t_res[o] = result["outputs"]
+        return flow
+
+    def evaluate(self, config, ee_id):
+        self._ee_dispach_url = config.get("dispatch_url")
+        self._ee_id = ee_id
+        self._eval_proc = multiprocessing.Process(
+            target=self._evaluate,
+            args=(self._ee_dispach_url, ee_id),
+        )
+        self._eval_proc.daemon = True
+        self._eval_proc.start()
+
+    def _evaluate(self, dispatch_url, ee_id):
```

Comment:
This function is long, could it be split into different  functions?

Reply:
yes :+1: